src/native/corehost/host_runtime_contract.h

```diff
@@ -53,4 +59,17 @@ struct host_runtime_contract
         const char* entry_point_name);
 };
 
+#if defined(TARGET_ANDROID)
+struct host_configuration_property
+{
+    const char16_t* name;
+    const char16_t* value;
+};
+
+struct host_configuration_properties
+{
+    size_t nitems;
+    const host_configuration_property* data;
+};
+#endif
```

Comment:
We have a few collections for properties. Why is this one needed?

/cc @elinor-fung

Reply:
It was part of the original contract, but the properties were passed as 3 parameters to `coreclr_initialize`.  Passing them as one pointer to a structure makes it use less registers on call and makes it easier to pass a `NULL` pointer to indicate there are no properties.  Also, `coreclr_initialize` expects UTF-8 properties, this structure uses Unicode.

Reply:
> Passing them as one pointer to a structure makes it use less registers on call and makes it easier to pass a NULL pointer to indicate there are no properties. Also, coreclr_initialize expects UTF-8 properties, this structure uses Unicode.

I really don't think we should be optimizing at this point. All of this is in service of getting things going and then we can work to create/design entry points that work. I get this is quick and dirty, but the quick and dirty should be on the consuming side when possible, not the runtime. There has been much talk about this API and how it is less than ideal, I don't think this PR is where we need to hack in a new one.

Reply:
This part can be removed then, however calling `coreclr_initialize` as it is now, with pointer-to-string-to-pointer conversions is not an option IMO, so we need to work around that aspect somehow. If we can modify `coreclr_initialize` to not expect the pointer-as-string abstraction, then sure, we can use it instead of the new API.